docs/deployment/overview.md

```diff
@@ -1,47 +1,295 @@
 
-- `aspirate init`: Initializes the **Aspir8** project in the current directory.
-- `aspirate generate`: Generates Kubernetes manifests based on the .NET Aspire AppHost manifest.
-- `aspirate apply`: Applies the generated Kubernetes manifests to the Kubernetes cluster.
-- `aspirate destroy`: Deletes the resources created by the `apply` command.
+```yaml
+services:
+  pg:
+    image: "docker.io/library/postgres:17.6"
+    environment:
+      POSTGRES_HOST_AUTH_METHOD: "scram-sha-256"
+      POSTGRES_INITDB_ARGS: "--auth-host=scram-sha-256 --auth-local=scram-sha-256"
+      POSTGRES_USER: "postgres"
+      POSTGRES_PASSWORD: "${PG_PASSWORD}"
+    ports:
+      - "8000:5432"
+    networks:
+      - "aspire"
+  dbsetup:
+    image: "${DBSETUP_IMAGE}"
+    environment:
+      OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EXCEPTION_LOG_ATTRIBUTES: "true"
+      OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES: "true"
+      OTEL_DOTNET_EXPERIMENTAL_OTLP_RETRY: "in_memory"
+      ASPNETCORE_FORWARDEDHEADERS_ENABLED: "true"
+      HTTP_PORTS: "8001"
+      ConnectionStrings__db: "Host=pg;Port=5432;Username=postgres;Password=${PG_PASSWORD};Database=db"
+    ports:
+      - "8002:8001"
+      - "8004:8003"
+    depends_on:
+      pg:
+        condition: "service_started"
+    networks:
+      - "aspire"
+```
 
-With these commands, you can build your apps, containerize them, and deploy them to Kubernetes clusters. For more information, see [Aspirate (formerly Aspir8)](https://prom3theu5.github.io/aspirational-manifests/).
+Key points:
 
-### Manually create Kubernetes manifests
+- `${PG_PASSWORD}` and `${DBSETUP_IMAGE}` (and similar) are *placeholders* in the published asset.
+- They are not resolved during `aspire publish`.
+- A deployment engine (which could be `aspire deploy`, `docker compose` plus a script exporting variables, CI/CD variable injection, etc.) supplies their values later.
+- This keeps secrets and environment-specific values decoupled from generated structure.
 
-Alternatively, the Kubernetes manifests can be created manually. This involves more effort and is more time consuming. For more information, see [Deploy a .NET microservice to Kubernetes](/training/modules/dotnet-deploy-microservices-kubernetes/).
+Different integrations may use different placeholder conventions (environment variables, tokens, or parameter metadata), but the principle remains: publish preserves *shape*, deploy injects *values*.
+
+## Publisher model and compute environments
+
+Aspire uses a flexible publisher model that distributes publishing behavior across your application graph. Resources support publishing and deployment through annotations:
+
+- `aspire publish`
+- `aspire deploy`
+
+This design enables hybrid and heterogeneous deployments, where different services within the same app can be deployed to different targets (cloud, edge, local).
```

Comment:
I think this statement hides some of the complexity here. We only support one of these annotatations per-resource at the moment so a hosting integration has to implement multiple resources to do the right thing. I think phrasing like:

> This annotation-based approach means that different hosting integrations within the same app can implement different publish and deployment behavior, supporting deployments to different targets.